modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/util/FilterURLCreatorUtil.java

```diff
@@ -0,0 +1,182 @@
+	}
+
+	public static String createFilterWithLambdaOperator(
+		String propertyName, String filter,
+		FilterOperator.LambdaOperator lambdaOperator) {
+
+		String lambdaContent = StringBundler.concat(
+			propertyName.substring(0, 1), StringPool.COLON, filter);
+
+		return _escape(
+			StringBundler.concat(
+				propertyName, StringPool.SLASH,
+				StringUtil.toLowerCase(lambdaOperator.name()),
+				_addParenthesis(lambdaContent)));
+	}
+
+	@SafeVarargs
+	public static <T> String createFilterWithListOperator(
+		FilterOperator.ListOperator listOperator, String propertyName,
+		T... propertyValues) {
+
+		StringBuilder propertyValuesStringBuilder = new StringBuilder(
+			StringPool.BLANK);
+
+		for (int i = 0; i < propertyValues.length; i++) {
+			if (i != (propertyValues.length - 1)) {
+				propertyValuesStringBuilder.append(
+					StringBundler.concat(
+						StringPool.APOSTROPHE, propertyValues[i],
+						StringPool.APOSTROPHE, StringPool.COMMA_AND_SPACE));
+			}
+			else {
+				propertyValuesStringBuilder.append(
+					StringBundler.concat(
+						StringPool.APOSTROPHE, propertyValues[i].toString(),
+						StringPool.APOSTROPHE));
+			}
+		}
+
+		return _escape(
+			StringBundler.concat(
+				propertyName, StringPool.SPACE,
+				StringUtil.toLowerCase(listOperator.name()), StringPool.SPACE,
+				_addParenthesis(propertyValuesStringBuilder.toString())));
+	}
+
+	public static String createFilterWithLogicalOperators(
+		String leftFilter, FilterOperator.LogicalOperator logicalOperator,
+		String rightFilter) {
+
+		if (logicalOperator == FilterOperator.LogicalOperator.NOT) {
+			return StringBundler.concat(
+				StringUtil.toLowerCase(logicalOperator.name()),
+				StringPool.SPACE, _addParenthesis(leftFilter));
+		}
+
+		return _escape(
+			StringBundler.concat(
+				leftFilter, StringPool.SPACE,
+				StringUtil.toLowerCase(logicalOperator.name()),
+				StringPool.SPACE, rightFilter));
+	}
+
+	public static String createFilterWithStringOperator(
+		String propertyName, String propertyValue,
+		FilterOperator.StringOperator stringOperator) {
+
+		String filter = StringBundler.concat(
+			propertyName, StringPool.COMMA_AND_SPACE, StringPool.APOSTROPHE,
+			propertyValue, StringPool.APOSTROPHE);
+
+		return _escape(
+			StringUtil.toLowerCase(stringOperator.name()) +
+				_addParenthesis(filter));
+	}
+
+	public static List<FilterOperator> getFilterOperators() {
+		List<FilterOperator> filterOperators = new ArrayList<>();
+
+		Collections.addAll(
+			filterOperators, FilterOperator.ComparisonOperator.values());
+		Collections.addAll(
+			filterOperators, FilterOperator.LambdaOperator.values());
+		Collections.addAll(
+			filterOperators, FilterOperator.ListOperator.values());
+		Collections.addAll(
+			filterOperators, FilterOperator.LogicalOperator.values());
+		Collections.addAll(
+			filterOperators, FilterOperator.StringOperator.values());
+
+		return filterOperators;
+	}
+
+	public interface FilterOperator {
```

Comment:
I am not sure if this make sense... why don't you just create every enum without implementing anything?